### ISSUE TITLE
Performance profiling

### ISSUE THREAD

**Issue by MarshalX** [MarshalX/python-libipld#7]
tldr:
- CIDs goes via multibase which a bit expensive
- ~Ipld allocated and dropped with short living time~ fixed in #8
- creaton of PyString, PyDict, calling set_item is expensive. pyo3 part
- decoding dag-cbor strings, bytes rust native libipld part

flame graph with debug (from #6):

[SVG direct link](https://raw.githubusercontent.com/gist/MarshalX/f4e5192e2ca210ece57239d04614c419/raw/3da59c21c62cb966781e47c99f87e95a13195cc0/flame.svg) ([Gist](https://gist.github.com/MarshalX/f4e5192e2ca210ece57239d04614c419#file-flame-svg))

![svg direct link](https://raw.githubusercontent.com/gist/MarshalX/f4e5192e2ca210ece57239d04614c419/raw/3da59c21c62cb966781e47c99f87e95a13195cc0/flame.svg)

**Comment by MarshalX** [comment]
new flame graph after fixes in #8: https://gist.github.com/MarshalX/542a87255bbd446dfafb6e74a517581c

**Comment by MarshalX** [comment]
codspeed has been integrated. flamegraph available for each pull request now.
